matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/permalinks/PermalinkService.kt

```diff
@@ -15,87 +15,64 @@
  * limitations under the License.
  */
 
-package org.matrix.android.sdk.api.permalinks
+package org.matrix.android.sdk.api.session.permalinks
 
 import org.matrix.android.sdk.api.session.events.model.Event
 
 /**
  * Useful methods to create Matrix permalink (matrix.to links).
  */
-object PermalinkFactory {
```

Comment:
I would have keep the PermalinkFactory as an internal class with required dependencies, and inject it on the permalink service to delegate. So we can use it internally in SDK without using the service, as we spoke. 
We also have a PermalinkParser, we might want to do the same.

Reply:
Yes, it will be better. Are you OK to merge this PR first and then schedule a global rework to remove all the injected services in the SDK?

Reply:
Ok, I've made the change for PermalinkFactory.
PermalinkParser is more static, so I prefer not touch it for the moment